numpy/tests/test_public_api.py

```diff
@@ -746,3 +747,64 @@ def test___module__attribute():
 
     if incorrect_entries:
         assert len(incorrect_entries) == 0, incorrect_entries
+
+
+def _check___qualname__(obj) -> bool:
+    qualname = obj.__qualname__
+    name = obj.__name__
+    module_name = obj.__module__
+    assert name == qualname.split(".")[-1]
+
+    module = sys.modules[module_name]
+    actual_obj = functools.reduce(getattr, qualname.split("."), module)
+    return (
+        actual_obj is obj or
+        (
+            # for bound methods check qualname match
+            module_name.startswith("numpy.random") and
+            actual_obj.__qualname__ == qualname
+        )
+    )
+
+
+def test___qualname___attribute():
+    modules_queue = [np]
+    visited_modules = {np}
+    visited_functions = set()
+    incorrect_entries = []
+
+    while len(modules_queue) > 0:
+        module = modules_queue.pop()
+        for member_name in dir(module):
+            member = getattr(module, member_name)
+            # first check if we got a module
+            if (
+                inspect.ismodule(member) and  # it's a module
+                "numpy" in member.__name__ and  # inside NumPy
+                not member_name.startswith("_") and  # not private
+                member_name not in [
+                    "f2py", "ma", "tests", "testing", "typing",
+                    "bit_generator", "ctypeslib", "lapack_lite",
+                ] and  # skip modules
```

Comment:
Let's follow up on it, some of these skips still look suspicious to me and I hope they originate from using `module_name` rather than `obj.__module__`.

Sorry, I didn't expect that you would just duplicated `show_config`, I'll say the same thing: I don't care either way, so let's just put this in.

Thanks for fixing the ufunc, it's nice to close that ancient issue and also thanks for making the test proper (which in the end found a regression after all!).